app/autosynccontroller.h

```diff
@@ -0,0 +1,68 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef AUTOSYNCCONTROLLER_H
+#define AUTOSYNCCONTROLLER_H
+
+#include <QObject>
+#include <QTimer>
+
+#include "qgsproject.h"
+
+#include "project.h"
+
+class AutosyncController : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY( SyncStatus syncStatus READ syncStatus NOTIFY syncStatusChanged )
+
+  public:
+
+    enum SyncStatus
+    {
+      Synced = 0,
```

Comment:
Synced == WaitingForNextSync? or means it will not try to sync anymore?

Reply:
I'd say it is a state when sync succeeded and there are no pending changes. These states, though, might change with implementation of the controller. It was more like a first try